scripts/cnode-helper-scripts/mithril.library

```diff
@@ -1,196 +1,196 @@
+ local env_file="${CNODE_HOME}/mithril/mithril.env"
+
+  if [[ -f "$env_file" ]]; then
+    if [[ "$UPDATE_ENVIRONMENT" != "Y" ]]; then
+      echo "Error: $env_file already exists. To update it, set UPDATE_ENVIRONMENT to 'Y'." >&2
+      return 1
+    else
+      echo "Updating $env_file..."
+    fi
+  else
+    echo "Creating $env_file..."
+  fi
+}
+
+get_relay_endpoint() {
+  read -r -p "Enter the IP address of the relay endpoint: " RELAY_ENDPOINT_IP
+  read -r -p "Enter the port of the relay endpoint (press Enter to use default 3132): " RELAY_PORT
+  RELAY_PORT=${RELAY_PORT:-3132}
+  echo "Using RELAY_ENDPOINT=${RELAY_ENDPOINT_IP}:${RELAY_PORT} for the Mithril signer relay endpoint."
+}
+
+update_mithril_environment_for_signer() {
+  echo "Info: Setting all environment variables, supporting the Mithril signer use case."
+  # Inquire about the relay endpoint
+  read -r -p "Are you using a relay endpoint? (y/n, press Enter to use default y): " ENABLE_RELAY_ENDPOINT
+  ENABLE_RELAY_ENDPOINT=${ENABLE_RELAY_ENDPOINT:-y}
+  if [[ "${ENABLE_RELAY_ENDPOINT}" == "y" ]]; then
+    get_relay_endpoint
+  else
+    echo "Using a naive Mithril configuration without a mithril relay."
+  fi
+  # Generate the full set of environment variables required by Mithril signer use case
+  export ERA_READER_ADDRESS=https://raw.githubusercontent.com/input-output-hk/mithril/main/mithril-infra/configuration/${RELEASE}-${NETWORK_NAME,,}/era.addr
+  export ERA_READER_VKEY=https://raw.githubusercontent.com/input-output-hk/mithril/main/mithril-infra/configuration/${RELEASE}-${NETWORK_NAME,,}/era.vkey
+  sudo bash -c "cat <<-'EOF' > ${CNODE_HOME}/mithril/mithril.env
+                KES_SECRET_KEY_PATH=${POOL_DIR}/${POOL_HOTKEY_SK_FILENAME}
+                OPERATIONAL_CERTIFICATE_PATH=${POOL_DIR}/${POOL_OPCERT_FILENAME}
+                NETWORK=${NETWORK_NAME,,}
+                RELEASE=${RELEASE}
+                AGGREGATOR_ENDPOINT=https://aggregator.${RELEASE}-${NETWORK_NAME,,}.api.mithril.network/aggregator
+                RUN_INTERVAL=60000
+                DB_DIRECTORY=${CNODE_HOME}/db
+                CARDANO_NODE_SOCKET_PATH=${CARDANO_NODE_SOCKET_PATH}
+                CARDANO_CLI_PATH=${HOME}/.local/bin/cardano-cli
+                DATA_STORES_DIRECTORY=${CNODE_HOME}/mithril/data-stores
+                STORE_RETENTION_LIMITS=5
+                ERA_READER_ADAPTER_TYPE=cardano-chain
+                ERA_READER_ADAPTER_PARAMS=$(jq -nc --arg address "$(wget -q -O - "${ERA_READER_ADDRESS}")" --arg verification_key "$(wget -q -O - "${ERA_READER_VKEY}")" '{"address": $address, "verification_key": $verification_key}')
+                GENESIS_VERIFICATION_KEY=$(wget -q -O - https://raw.githubusercontent.com/input-output-hk/mithril/main/mithril-infra/configuration/${RELEASE}-${NETWORK_NAME,,}/genesis.vkey)
+                PARTY_ID=$(cat ${POOL_DIR}/${POOL_ID_FILENAME}-bech32)
+                SNAPSHOT_DIGEST=latest
+                ENABLE_METRICS_SERVER=true

```

Comment:
1. Should make this a choice of user input, like ENABLE_RELAY_ENDPOINT from lines [135-138](https://github.com/cardano-community/guild-operators/pull/1786/files#diff-222307d8e0610ac57be518231d9ac3fa96991b9da929d4f314937c0fd7e19eaaR135-R138) and [lines 165-167](https://github.com/cardano-community/guild-operators/pull/1786/files#diff-222307d8e0610ac57be518231d9ac3fa96991b9da929d4f314937c0fd7e19eaaR165-R167)

2. The `METRICS_SERVER_PORT` should also be a choice of user input. Otherwise if the signer runs on the same system as the monitoring is configured it would conflict, [setup_mon.sh Line 13](https://github.com/cardano-community/guild-operators/blob/alpha/scripts/cnode-helper-scripts/setup_mon.sh#L13) defaults to the same port.

3. Also `METRICS_SERVER_IP` from user input. This way the signer metric server does not automatically bind to all addresses if the user would prefer it not to. Also provides an option to keep the signer metric server port 9090, but but a different interface/IP than the `PROM_PORT`.